compiler/packages/babel-plugin-react-compiler/src/HIR/HIRBuilder.ts

```diff
@@ -271,9 +271,14 @@ export default class HIRBuilder {
           module: importDeclaration.node.source.value,
         };
       } else {
+        const immutable =
```

Comment:
Hmm, apparently you can reassign to bindings created by class / function declarations. Not sure if this is in the spec but it's allowed by node

```js
class x {}
x = 2;
console.log(x);
```

Reply:
Thanks for checking me on this! Yeah, I think we can restrict it to just const.